libcloudforensics/providers/azure/internal/account.py

```diff
@@ -66,8 +66,11 @@ def __init__(self,
     self._network = None  # type: Optional[network_module.AZNetwork]
     self._resource = None  # type: Optional[resource_module.AZResource]
     self._storage = None  # type: Optional[storage_module.AZStorage]
-    self.default_resource_group_name = self.resource.GetOrCreateResourceGroup(
-        default_resource_group_name)
+    if default_resource_group_name:
+      self.default_resource_group_name = self.resource.GetOrCreateResourceGroup(
+          default_resource_group_name)
+    else:
+      self.default_resource_group_name = ''
```

Comment:
I think here what we could do is make `GetOrCreateResourceGroup` return `''` if the given resource name is `''`. This way, you can rewrite `init()` as:

```python
  def __init__(self,
               default_resource_group_name: str = '',
               default_region: str = 'eastus',
               profile_name: Optional[str] = None) -> None:
```

And assign the default resource name as:

```python
self.default_resource_group_name = self.resource.GetOrCreateResourceGroup(default_resource_group_name)
```

Without the `if/else` block. You just need to add the following block to `GetOrCreateResourceGroup` at its beginning:

```python
if not resource_group_name:
  return ''
```

Reply:
Done